lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -3339,28 +3337,26 @@ constructSharedTransaction
                             []
                         Just (ApiPaymentAddresses content) ->
                             F.toList (addressAmountToTxOut <$> content)
-                (unbalancedTx, scriptLookup) <- liftHandler $
+                unbalancedTx <- liftHandler $
                     W.constructUnbalancedSharedTransaction @n era
                     db txCtx PreSelection {outputs = outs}
 
-                balancedTx <-
-                    balanceTransaction api argGenChange
-                    (AllScriptPaymentCredentialsFrom
-                        (Shared.paymentTemplate (getState cp))
-                        (scriptLookup . Convert.toWalletAddress)
-                    )
-                    mempty
-                    (ApiT wid)
-                        ApiBalanceTransactionPostData
-                        { transaction =
-                            ApiT $ sealedTxFromCardanoBody unbalancedTx
-                        , inputs = []
-                        , redeemers = []
-                        , encoding = body ^. #encoding
+                balancedTx <- liftIO $ W.balanceTx
+                    wrk
+                    pp
+                    timeTranslation
+                    PartialTx
+                        { tx = Write.fromCardanoApiTx
+                             $ Cardano.Tx unbalancedTx []
+                        , extraUTxO = mempty
+                        , redeemers = mempty
+                        , timelockKeyWitnessCounts = mempty
                         }
 
                 apiDecoded <- decodeSharedTransaction api (ApiT wid)
-                              (toApiDecodeTransactionPostData balancedTx)
+                    $ ApiDecodeTransactionPostData
+                        (ApiT (sealWriteTx balancedTx))
+                        Nothing
```

Comment:
Suggestion: use explicit record-style construction here, especially if it will make it clearer to the reader what `Nothing` refers to without requiring them to visit the definition of `ApiDecodeTransactionPostData`:
```hs
ApiDecodeTranscationPostData
    { sealedTx = ApiT (sealWriteTx balancedTx)
    , mysteryField = Nothing
    }
```